### PR TITLE
Minimize Nuclei DAST PoC URLs

### LINKED ISSUE #1977: Somehow minimize the PoC URLs generated by the Nuclei DAST templates as they're big
`/vulnerabilities/sqli.php?dest=http%3A%2F%2F127.0.0.1%2Fabc.html'&redirect=http%3A%2F%2F127.0.0.1%2Fabc.html'&uri=http%3A%2F%2F127.0.0.1%2Fabc.html'&path=http%3A%2F%2F127.0.0.1%2Fabc.html'&continue=http%3A%2F%2F127.0.0.1%2Fabc.html'&url=http%3A%2F%2F127.0.0.1%2Fabc.html'&window=http%3A%2F%2F127.0.0.1%2Fabc.html'&next=http%3A%2F%2F127.0.0.1%2Fabc.html'&data=http%3A%2F%2F127.0.0.1%2Fabc.html'&reference=http%3A%2F%2F127.0.0.1%2Fabc.html'&site=http%3A%2F%2F127.0.0.1%2Fabc.html'&html=http%3A%2F%2F127.0.0.1%2Fabc.html'&val=http%3A%2F%2F127.0.0.1%2Fabc.html'&validate=http%3A%2F%2F127.0.0.1%2Fabc.html'&domain=http%3A%2F%2F127.0.0.1%2Fabc.html'&callback=http%3A%2F%2F127.0.0.1%2Fabc.html'&return=http%3A%2F%2F127.0.0.1%2Fabc.html'&page=http%3A%2F%2F127.0.0.1%2Fabc.html'&feed=http%3A%2F%2F127.0.0.1%2Fabc.html'&host=http%3A%2F%2F127.0.0.1%2Fabc.html'&port=http%3A%2F%2F127.0.0.1%2Fabc.html'&to=http%3A%2F%2F127.0.0.1%2Fabc.html'&out=http%3A%2F%2F127.0.0.1%2Fabc.html'&view=http%3A%2F%2F127.0.0.1%2Fabc.html'&dir=http%3A%2F%2F127.0.0.1%2Fabc.html'&target=http%3A%2F%2F127.0.0.1%2Fabc.html'&rurl=http%3A%2F%2F127.0.0.1%2Fabc.html'&destination=http%3A%2F%2F127.0.0.1%2Fabc.html'&redir=http%3A%2F%2F127.0.0.1%2Fabc.html'&redirect_uri=http%3A%2F%2F127.0.0.1%2Fabc.html'&image_url=http%3A%2F%2F127.0.0.1%2Fabc.html'&go=http%3A%2F%2F127.0.0.1%2Fabc.html'&returnTo=http%3A%2F%2F127.0.0.1%2Fabc.html'&return_to=http%3A%2F%2F127.0.0.1%2Fabc.html'&checkout_url=http%3A%2F%2F127.0.0.1%2Fabc.html'&return_path=http%3A%2F%2F127.0.0.1%2Fabc.html'&cat=abc.html'&action=abc.html'&board=abc.html'&date=abc.html'&detail=abc.html'&file=abc.html'&filename=abc.html'&download=abc.html'&folder=abc.html'&prefix=abc.html'&include=abc.html'&inc=abc.html'&locate=abc.html'&show=abc.html'&doc=abc.html'&type=abc.html'&content=abc.html'&document=abc.html'&layout=abc.html'&mod=abc.html'&conf=abc.html'&cmd=testing'&exec=testing'&command=testing'&execute=testing'&ping=testing'&query=testing'&jump=testing'&code=testing'®=testing'&do=testing'&func=testing'&arg=testing'&option=testing'&load=testing'&process=testing'&step=testing'&read=testing'&function=testing'&req=testing'&feature=testing'&exe=testing'&module=testing'&payload=testing'&run=testing'&print=testing'&id=testing'&search=testing'&category=testing'&class=testing'&news=testing'&item=testing'&menu=testing'&lang=testing'&name=testing'&ref=testing'&title=testing'&topic=testing'&thread=testing'&form=testing'&join=testing'&main=testing'&nav=testing'®ion=testing'&q=testing'&s=testing'&keyword=testing'&keywords=testing'&year=testing'&email=testing'&p=testing'&month=testing'&image=testing'&list_type=testing'&terms=testing'&categoryid=testing'&key=testing'&login=testing'&begindate=testing'&enddate=testing`

### PR DESCRIPTION
**Pull request by Sumit-ai-dev**
Fixes #1977 by minimizing the excessively long PoC URLs generated by Nuclei DAST templates.

Instead of including 100+ filler parameters in the vulnerability report, this uses Nuclei's `fuzzing_parameter` JSON field to isolate exactly which parameter was vulnerable and drops the rest. 

As requested in the issue, the logic is written defensively:
- **If `fuzzing_parameter` is found:** We keep only the vulnerable parameter (e.g. `?id=testing'`).
- **If not (fallback):** We return the full original URL unchanged to guarantee the PoC remains functional.

### Changes
- **`poc_url_utils.py`:** Added the core `minimize_nuclei_matched_at_url` function with the defensive fallback.
- **`reporter.py`:** Wired up the new function and cleaned up obsolete filler parameters.
- **Tests:** Added 8 unit tests covering all edge cases, the defensive fallback, and the exact SQLi payload from the issue.

### CODE REVIEW

File: artemis/reporting/modules/nuclei/poc_url_utils.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+"""Utilities for minimizing Nuclei DAST PoC URLs."""
+
+import urllib.parse
+from typing import Optional
+
+
+def minimize_nuclei_matched_at_url(
```
Comment (kazet) [inline artemis/reporting/modules/nuclei/poc_url_utils.py:18]: I see one thing missing - checking, that the resulting URL contains the vulnerability. We can have rare cases where the single-parameter URL doesn't work as multiple needs to be set (as the app checks them via isset()). In these cases we need to fallback to the original URL.

If this one gets fixed, we can get this PR merged.

Comment (Sumit-ai-dev) [reply artemis/reporting/modules/nuclei/poc_url_utils.py:18]: Added a check to make a GET request to the minimized URL. If it doesn't return 200, it falls back to the original. Just pushed!

Comment (kazet) [reply artemis/reporting/modules/nuclei/poc_url_utils.py:18]: What about re-running this particular Nuclei template? I know it's harder but will decrease the number of FPs

Comment (Sumit-ai-dev) [reply artemis/reporting/modules/nuclei/poc_url_utils.py:18]: I updated the logic to spawn the [nuclei](cci:1://file:///Users/sumitdas/Desktop/Artemis/Artemis/artemis/reporting/modules/nuclei/reporter.py:45:4-70:24) CLI in the background and actually re-run the template against the minimized URL.

I used the exact same flags and command structure from [artemis/modules/nuclei.py](cci:7://file:///Users/sumitdas/Desktop/Artemis/Artemis/artemis/modules/nuclei.py:0:0-0:0) to keep it consistent. If it doesn't trigger anymore (like if an `isset()` param was dropped), it safely falls back to the original URL.

Just pushed the update!

File: artemis/reporting/modules/nuclei/reporter.py [hunk 2]
```diff
@@ -38,6 +43,33 @@ class NucleiReporter(Reporter):
...
+                "-t",
+                os.path.join(NUCLEI_TEMPLATES_LOCATION, template_id),
+                "-dast",
+                "-fuzzing-mode",
+                "multiple",
+                "-jsonl",
+                "-silent",
+            ]
+            result = subprocess.check_output(command, stderr=subprocess.DEVNULL)
+            return bool(result.strip())
```
Comment (kazet) [inline artemis/reporting/modules/nuclei/reporter.py]: I'm not sure checking that the result is nonempty is enough, could you add a little more checks?

File: artemis/reporting/modules/nuclei/reporter.py [hunk 3]
```diff
@@ -38,6 +43,33 @@ class NucleiReporter(Reporter):
...
+        except FileNotFoundError:
+            # If nuclei isn't available (e.g. in some local dev envs), fail open
+            # to preserve the minimized URL so reporting doesn't break.
+            logging.getLogger(__name__).warning("Nuclei binary not found, skipping URL verification")
```
Comment (kazet) [inline artemis/reporting/modules/nuclei/reporter.py]: we have self.log ;)

File: artemis/reporting/modules/nuclei/reporter.py [hunk 4]
```diff
@@ -38,6 +43,33 @@ class NucleiReporter(Reporter):
...
+            return True
```
Comment (kazet) [inline artemis/reporting/modules/nuclei/reporter.py]: IMO we should return False to be consistent with the comment

File: artemis/reporting/modules/nuclei/reporter.py [hunk 5]
```diff
@@ -145,7 +177,13 @@ def _is_url_without_query_fragment(url: str) -> bool:
                 #
                 # We want to restrict this behavior only to services on empty path (e.g. redis://some-domain:6379) because
                 # if the path is nonempty, it may contain an exploit that may get filtered by e-mail filters.
-                matched_at = add_protocol_if_needed(vulnerability.get("matched-at", target))
+                matched_at = minimize_nuclei_matched_at_url(
```
Comment (kazet) [inline artemis/reporting/modules/nuclei/reporter.py]: Could you add an end-to-end for that? Along the lines of the tests in tests/modules/nuclei.py

File: artemis/reporting/modules/nuclei/reporter.py (hunk 2 above)
Comment (Sumit-ai-dev) [reply artemis/reporting/modules/nuclei/reporter.py]: Fixed now parsing each JSONL line with json.loads() and only returning True if matched-at is present in the parsed output. Avoids false positives from verbose stdout noise.

File: artemis/reporting/modules/nuclei/reporter.py (hunk 3 above)
Comment (Sumit-ai-dev) [reply artemis/reporting/modules/nuclei/reporter.py]: _verify_with_nuclei
 is a @staticmethod so self isn't available here. Added a module-level logger = logging.getLogger(__name__) instead — same effect, no more inline getLogger() calls.

File: artemis/reporting/modules/nuclei/reporter.py (hunk 4 above)
Comment (Sumit-ai-dev) [reply artemis/reporting/modules/nuclei/reporter.py]: Done, changed to return False.

File: artemis/reporting/modules/nuclei/reporter.py (hunk 5 above)
Comment (Sumit-ai-dev) [reply artemis/reporting/modules/nuclei/reporter.py]: Added  test_dast_matched_at_url_is_minimized in 
test_nuclei_autoreporter_integration.py— uses obtain_http_task_result against test-dast-vuln-app and asserts matched_ at URLs have ≤ 1 query parameter after minimization.

**Comment by Sumit-ai-dev** [comment]
@kazet Addressed all your concerns, JSONL parsing with matched-at check, module-level logger, return False on missing binary, and the end-to-end test. Just pushed. Happy to adjust anything further. :)

**Comment by kazet** [comment]
The test didn't pass, could you fix it?

**Comment by Sumit-ai-dev** [comment]
@kazet Sure, just pushed the fix. Should pass now!

**Comment by kazet** [comment]
Would it require much changes to also make it work on linux-lfi-fuzz?

**Comment by Sumit-ai-dev** [comment]
Not much when fuzzing_parameter is absent, we could iterate each query param individually and verify each one to find which one triggers the vulnerability. Happy to implement once this PR merges. I can open an issue and raise a separate PR for it, or just raise a PR directly. Whatever you prefer, @kazet.

**Comment by Sumit-ai-dev** [comment]
Turns out lfi-keyed doesn't trigger on the test app  only linux-lfi-fuzz fires. So I implemented the per-param iteration instead iterates each param individually, verifying each one, returns whichever triggers. Just pushed. :)

File: artemis/reporting/modules/nuclei/reporter.py [hunk 6]
```diff
@@ -38,6 +48,38 @@ class NucleiReporter(Reporter):
     with open(Config.Modules.Nuclei.NUCLEI_TEMPLATE_GROUPS_FILE) as f:
         GROUPS = json.load(f)
 
+    @staticmethod
+    def _verify_with_nuclei(url: str, template_id: str) -> bool:
```
Comment (kazet) [inline artemis/reporting/modules/nuclei/reporter.py]: One minor comment - I think it would be optimal to do it during `scanning`, not `report generation`, so that e.g. generating multiple reports doesn't run multiple minimizations. After that we are good to go!

Comment (Sumit-ai-dev) [reply artemis/reporting/modules/nuclei/reporter.py]: Done moved the minimization to nuclei.py at scan time. 
_verify_with_nuclei
 is now a module-level function there, and matched-at gets minimized before saving to DB. Reporter just reads the already-minimized value. Pushed.

**Comment by kazet** [comment]
if the tests pass, we can get it merged!
